### PR TITLE
FIX validation failure without instance features

### LINKED ISSUE #377: Validation fails if instances without features are present
As described [here](https://github.com/automl/CAVE/issues/114).

### PR DESCRIPTION
**Pull request by shukon**
fixes #377, adds test

### CODE REVIEW

File: smac/utils/validate.py [hunk 1]
```diff
@@ -308,13 +308,14 @@ def validate_epm(self,
...
-        try:
+        if hasattr(self.scen, "feature_array") and self.scen.feature_array is not None:
             feature_array_size = len(self.scen.cs.get_hyperparameters()) + self.scen.feature_array.shape[1]
-        except AttributeError:
+        else:
             feature_array_size = len(self.scen.cs.get_hyperparameters())
         X_pred = np.empty((len(runs), feature_array_size))
         for idx, run in enumerate(runs):
-            if hasattr(self.scen, "feature_dict") and run.inst != None:
+            if (hasattr(self.scen, "feature_array") and
```
Comment (mfeurer) [inline smac/utils/validate.py]: This looks quite error-prone. Would it be possible to always assign a `feature_array` to the scenario and only check if it is `None` here?

File: test/test_utils/test_validate.py [hunk 2]
```diff
@@ -335,3 +335,22 @@ def test_objective_runtime(self):
...
             old_rh.add(config, 1, 1, StatusType.SUCCESS, instance_id='0')
         validator.validate_epm('all', 'train', 1, old_rh)
+
+    def test_inst_no_feat(self):
+        ''' test if scenarios are treated correctly if no features are
+        specified.'''
+        scen = Scenario(self.scen_fn,
+                        cmd_args={'run_obj':'quality',
+                                  'instances' : self.train_insts,
+                                  'test_instances': self.test_insts})
```
Comment (mfeurer) [inline test/test_utils/test_validate.py]: Could you please add some assert that no features are present?

File: smac/utils/validate.py (hunk 1 above)
Comment (shukon) [reply smac/utils/validate.py]: Absolutely. Actually `Scenario` is already doing that and I do not remember any other reason for using `hasattr`, so I must have missed it when writing the original code.

File: test/test_utils/test_validate.py (hunk 2 above)
Comment (shukon) [reply test/test_utils/test_validate.py]: Done

**Pull request merged**